File: web_front/src/lib/api/resume.ts
/** 简历模块API客户端，对接后端 /api/v1/resumes 端点。 */

import request from '@/lib/request';

/** 简历解析状态（对齐后端 resume.status） */
export type ResumeStatus = 0 | 1 | 2;

/** 工作经历（对齐后端 ResumeWorkExperienceOut） */
export interface ApiWorkExperience {
  id: number;
  company: string;
  role: string;
  duration: string | null;
  description: string | null;
  sort_order: number;
}

/** 后端简历详情/列表项（对齐后端 ResumeOut） */
export interface ApiResume {
  id: number;
  user_id: number;
  file_name: string;
  file_url: string | null;
  file_size: number | null;
  status: ResumeStatus;
  parsed_name: string | null;
  parsed_skills: string[] | null;
  parsed_education: Array<Record<string, unknown>> | null;
  parsed_projects: Array<Record<string, unknown>> | null;
  error_message: string | null;
  created_at: string;
  updated_at: string;
  work_experiences: ApiWorkExperience[];
}

/** 简历分页列表响应 */
export interface ApiResumeListResponse {
  items: ApiResume[];
  total: number;
  page: number;
  page_size: number;
}

/** 解析状态文案/配色映射 */
export const RESUME_STATUS_LABEL: Record<ResumeStatus, string> = {
  0: '解析中',
  1: '已就绪',
  2: '解析失败',
};

/**
 * 查询当前用户简历列表（分页）。
 * @param page 页码（从1开始）
 * @param pageSize 页大小
 */
export async function getResumes(page = 1, pageSize = 20): Promise<ApiResumeListResponse> {
  const { data } = await request.get<ApiResumeListResponse>('/resumes', {
    params: { page, page_size: pageSize },
  });
  return data;
}

/**
 * 删除简历（软删 + 联动清理COS/缓存/锁）。
 * @param resumeId 简历ID
 */
export async function deleteResume(resumeId: number): Promise<void> {
  await request.delete(`/resumes/${resumeId}`);
}

/**
 * 对解析失败的简历一键重试。
 * @param resumeId 简历ID
 */
export async function retryResume(resumeId: number): Promise<ApiResume> {
  const { data } = await request.post<ApiResume>(`/resumes/${resumeId}/retry`);
  return data;
}

File: api_server/app/schemas/resume.py
"""简历相关请求/响应模型。"""

from datetime import datetime

from pydantic import BaseModel, Field


class ResumeWorkExperienceOut(BaseModel):
    """简历工作经历输出模型。"""

    id: int = Field(..., description="工作经历ID")
    company: str = Field(..., description="公司名称")
    role: str = Field(..., description="职位")
    duration: str | None = Field(None, description="任职时间")
    description: str | None = Field(None, description="工作描述")
    sort_order: int = Field(0, description="排序")

    model_config = {"from_attributes": True}


class ResumeOut(BaseModel):
    """简历输出模型（列表项/详情）。"""

    id: int = Field(..., description="简历ID")
    user_id: int = Field(..., description="所属用户ID")
    file_name: str = Field(..., description="原始文件名")
    file_url: str | None = Field(None, description="文件访问URL")
    file_size: int | None = Field(None, description="文件大小（字节）")
    status: int = Field(..., description="解析状态 0-解析中 1-就绪 2-错误")
    parsed_name: str | None = Field(None, description="解析出的姓名")
    parsed_skills: list[str] | None = Field(None, description="解析出的技能列表")
    parsed_education: list[dict] | None = Field(None, description="解析出的教育经历数组")
    parsed_projects: list[dict] | None = Field(None, description="解析出的项目经历数组")
    error_message: str | None = Field(None, description="解析失败原因")
    created_at: datetime = Field(..., description="创建时间")
    updated_at: datetime = Field(..., description="更新时间")
    work_experiences: list[ResumeWorkExperienceOut] = Field(
        default_factory=list, description="工作经历列表"
    )

    model_config = {"from_attributes": True}


class ResumeListResponse(BaseModel):
    """简历列表响应模型（分页）。"""

    items: list[ResumeOut] = Field(..., description="简历列表")
    total: int = Field(..., description="简历总数")
    page: int = Field(..., description="当前页码")
    page_size: int = Field(..., description="页大小")
